wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/ApiConnection.java

```diff
@@ -86,48 +74,55 @@
 	 * Name of the HTTP parameter to submit the requested result format to the
 	 * API.
 	 */
-	public final static String PARAM_FORMAT = "format";
+	protected final static String PARAM_FORMAT = "format";
 
 	/**
-	 * MediaWiki assert= parameter to ensure we are editting while logged in.
+	 * MediaWiki assert parameter to ensure we are editing while logged in.
 	 */
-	private static final String ASSERT_PARAMETER = "assert";
+	protected static final String ASSERT_PARAMETER = "assert";
+
+	protected static final MediaType URLENCODED_MEDIA_TYPE = MediaType.parse("application/x-www-form-urlencoded");
 
 	/**
 	 * URL to access the Wikibase API.
 	 */
-	final String apiBaseUrl;
+	protected final String apiBaseUrl;
 
 	/**
 	 * True after successful login.
 	 */
-	boolean loggedIn = false;
+	protected boolean loggedIn = false;
 	/**
 	 * User name used to log in.
 	 */
-	String username = "";
-	
+	protected String username = "";
+
 	/**
 	 * Map of requested tokens.
 	 */
-	final Map<String, String> tokens;
-	
+	protected final Map<String, String> tokens;
+
 	/**
 	 * Maximum time to wait for when establishing a connection, in milliseconds.
 	 * For negative values, no timeout is set.
 	 */
-	int connectTimeout = -1;
-	
+	protected int connectTimeout = -1;
```

Comment:
We could maybe set most of the fields here private. This way we are sure that the subclasses are using the getter/setters.

Reply:
We can't use private here, because this field is needed for the json creator.

Reply:
For that you could use the `@JsonGetter` annotation on the getter methods.